Azure/sonic-snmpagent
# [multi-asic][RFC1213]: Oper status of mgmt interface for multi-asic platform

**- What I did**
Currently oper status of all management interface is retrieved from STATE DB.
Multi-asic platform does not have any daemon running on the host namespace to update oper status of management interface in STATE DB. This PR is to add a function to retrieve oper status of management interface from kernel file: "/sys/class/net/if_name/operstate".  
**- How I did it**
- Add a new function to retrieve oper status from /sys/class/net/if_name/operstate.
- Invoke this function periodically to retrieve oper status of all management interfaces.
- This is done periodically, instead of retrieving only on SNMP query, so that the time take during the SNMP query does not increase due to the file read operation.
- Add unit-tests for mgmt interface.
**- How to verify it**
- Run snmpwalk to get interface status, ensure that the interface status of management interface is correct, on single asic and multi-asic VS testbed.
- With the changes - no impact of single asic platform.
multi-asic VS testbed:
```
sudo ifconfig eth0 down
docker exec -it snmp snmpwalk -v2c -c public 127.0.0.1 1.3.6.1.2.1.2.2.1.8
iso.3.6.1.2.1.2.2.1.8.1 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.13 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.17 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.21 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.25 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.29 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1001 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1002 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1003 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1004 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1005 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1006 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5002 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5003 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5012 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5013 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5020 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5021 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5030 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5031 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9000 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9004 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9008 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9012 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9016 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9020 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9024 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9028 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9032 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9036 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9040 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9044 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9048 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9052 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9056 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9060 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9064 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9068 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9072 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9076 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9080 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9084 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9092 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9096 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10000 = INTEGER: 2 --> status down
sudo ifconfig eth0 up
docker exec -it snmp snmpwalk -v2c -c public 127.0.0.1 1.3.6.1.2.1.2.2.1.8
iso.3.6.1.2.1.2.2.1.8.1 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.13 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.17 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.21 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.25 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.29 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1001 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1002 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1003 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1004 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1005 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.1006 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5002 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5003 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5012 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5013 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5020 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5021 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5030 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.5031 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9000 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9004 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9008 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9012 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9016 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9020 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9024 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9028 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9032 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9036 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9040 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9044 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9048 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9052 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9056 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9060 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9064 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9068 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9072 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9076 = INTEGER: 1
iso.3.6.1.2.1.2.2.1.8.9080 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9084 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9092 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.9096 = INTEGER: 2
iso.3.6.1.2.1.2.2.1.8.10000 = INTEGER: 1 --> status up

**- Description for the changelog**
